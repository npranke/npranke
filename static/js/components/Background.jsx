import React from 'react'

import Landscape from '@img/background-landscape.jpg'
import Portrait from '@img/background-portrait.jpg'

function Background() {
    return (
        <picture>
            <source srcSet={ Portrait } media="(orientation: portrait)" />
            <img
                src={ Landscape }
                className="background"
                alt="Conifer in Colorado landscape"
            />
        </picture>
    )
}

export default Background
