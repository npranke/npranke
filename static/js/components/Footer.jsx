function Footer() {
    const currentYear = new Date().getFullYear()

    return (
        <footer className="footer" role="contentinfo">
            <div className="copyright">
                &copy; 2017-{currentYear} Nicole Pranke
            </div>
        </footer>
    )
}

export default Footer
