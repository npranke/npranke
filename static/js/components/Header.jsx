import { NavLink } from 'react-router-dom-v5-compat'
import PropTypes from 'prop-types'
import React from 'react'

import GitHub from '@img/icon-github.png'
import Home from '@img/icon-home.png'
import LinkedIn from '@img/icon-linkedin.png'
import Workbook from '@img/icon-workbook.png'

import { getSendEventHandler } from '@utils'

function Header(props) {
    const homeLink = props.isPageNotFound
        ? (
            <a
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'home',
                ) }
                href="/home"
                target="_self"
                rel="noreferrer"
                title="Home"
                aria-label="Home"
            >
                <img src={ Home } className="icon icon-home" alt="Home icon" />
            </a>
        ) : (
            <NavLink
                end
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'home',
                ) }
                to="/home"
                target="_self"
                rel="noreferrer"
                className="navlink navlink-home"
                title="Home"
                aria-label="Home"
            >
                <img src={ Home } className="icon icon-home" alt="Home icon" />
            </NavLink>
        )

    const workbookLink = props.isPageNotFound
        ? (
            <a
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'workbook',
                ) }
                href="/workbook"
                target="_self"
                rel="noreferrer"
                title="Workbook"
                aria-label="Workbook"
            >
                <img
                    src={ Workbook }
                    className="icon icon-workbook"
                    alt="Workbook icon"
                />
            </a>
        ) : (
            <NavLink
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'workbook',
                ) }
                to="/workbook"
                target="_self"
                rel="noreferrer"
                className="navlink navlink-workbook"
                title="Workbook"
                aria-label="Workbook"
            >
                <img
                    src={ Workbook }
                    className="icon icon-workbook"
                    alt="Workbook icon"
                />
            </NavLink>
        )

    return (
        <nav className="header">
            <div className="left-header">
                <div className="icon-container icon-container-home">
                    { homeLink }
                </div>
            </div>
            <div className="right-header">
                <div className="icon-container icon-container-workbook">
                    { workbookLink }
                </div>
                <div className="icon-container icon-container-github">
                    <a
                        onClick={ getSendEventHandler(
                            'header',
                            'navigate',
                            'github',
                        ) }
                        href="https://github.com/npranke"
                        target="_blank"
                        rel="noreferrer noopener external"
                        title="GitHub"
                        aria-label="GitHub"
                    >
                        <img
                            src={ GitHub }
                            className="icon icon-github"
                            alt="GitHub icon"
                        />
                    </a>
                </div>
                <div className="icon-container icon-container-linkedin">
                    <a
                        onClick={ getSendEventHandler(
                            'header',
                            'navigate',
                            'linkedin',
                        ) }
                        href="https://www.linkedin.com/in/npranke"
                        target="_blank"
                        rel="noreferrer noopener external"
                        title="LinkedIn"
                        aria-label="LinkedIn"
                    >
                        <img
                            src={ LinkedIn }
                            className="icon icon-linkedin"
                            alt="LinkedIn icon"
                        />
                    </a>
                </div>
            </div>
        </nav>
    )
}

Header.defaultProps = {
    isPageNotFound: false,
}

Header.propTypes = {
    isPageNotFound: PropTypes.bool,
}

export default Header
