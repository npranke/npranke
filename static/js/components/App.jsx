import React from 'react'
import { Route, Switch } from 'react-router-dom'

import Background from './Background'
import Footer from './Footer'
import Header from './Header'
import PageNotFound from './PageNotFound'
import Welcome from './Welcome'
import Workbook from './Workbook'
import WorksheetContainer from './WorksheetContainer'
import worksheets from '../constants/worksheets'

function App() {
    const { concentration } = worksheets

    return (
        <div className="app">
            <Background />
            <Switch>
                <Route
                    exact
                    strict
                    path={ [
                        '/',
                        '/home',
                        '/workbook',
                        '/workbook/concentration',
                    ] }
                    component={ Header }
                />
                <Route
                    render={ (props) => {
                        return <Header { ...props } isPageNotFound />
                    } }
                />
            </Switch>
            <Switch>
                <Route
                    exact
                    strict
                    path={ ['/', '/home'] }
                    component={ Welcome }
                />
                <Route
                    exact
                    strict
                    path="/workbook"
                    component={ Workbook }
                />
                <Route
                    exact
                    strict
                    path="/workbook/concentration"
                    render={ (props) => {
                        return (
                            <WorksheetContainer
                                { ...props }
                                worksheet={ concentration }
                            />
                        )
                    } }
                />
                <Route component={ PageNotFound } />
            </Switch>
            <Footer />
        </div>
    )
}

export default App
