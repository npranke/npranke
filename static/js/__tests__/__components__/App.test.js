import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import App from '../../components/App'
import Background from '../../components/Background'
import Footer from '../../components/Footer'
import Header from '../../components/Header'

Enzyme.configure({ adapter: new Adapter() })

describe('App', () => {
    test('contains background', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Background />),
        ).toBe(true)
    })

    test('contains header', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Header />),
        ).toBe(true)
    })

    test('contains switch', () => {
        const app = shallow(<App />)

        expect(
            app.exists('Switch'),
        ).toBe(true)
    })

    test('contains routes', () => {
        const app = shallow(<App />)

        expect(
            app.find('Route'),
        ).toHaveLength(2)
    })

    test('contains footer', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Footer />),
        ).toBe(true)
    })
})

describe('App snapshot', () => {
    test('matches snapshot', () => {
        const app = shallow(<App />)

        expect(app).toMatchSnapshot()
    })
})
