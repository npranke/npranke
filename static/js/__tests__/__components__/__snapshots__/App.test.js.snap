// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot matches snapshot 1`] = `
<div
  className="app"
>
  <Background />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path={
        Array [
          "/",
          "/home",
          "/workbook",
          "/workbook/concentration",
        ]
      }
      strict={true}
    />
    <Route
      render={[Function]}
    />
  </Switch>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path={
        Array [
          "/",
          "/home",
        ]
      }
      strict={true}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/workbook"
      strict={true}
    />
    <Route
      exact={true}
      path="/workbook/concentration"
      render={[Function]}
      strict={true}
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <Footer />
</div>
`;
